src/engine/engine_common.c

```diff
@@ -636,6 +649,45 @@ void ocf_engine_push_req_front(struct ocf_request *req, bool allow_sync)
 	ocf_queue_kick(q, allow_sync);
 }
 
+void ocf_engine_reschedule_deferred(ocf_queue_t q)
+{
+	struct ocf_request *curr, *temp_req;
+	unsigned long lock_flags = 0;
+
+	if (env_atomic_read(&q->deferred_io_no) == 0)
+		return;
+
+	env_spinlock_lock_irqsave(&q->deferred_io_list_lock, lock_flags);
+
+	list_for_each_entry_safe(curr, temp_req, &q->deferred_io_list, list) {
+		list_del(&curr->list);
+		env_atomic_dec(&q->deferred_io_no);
+		ocf_engine_push_req_front(curr, false);
+	}
```

Comment:
I think it might make sense to add a function to put entire list content at the beginning of I/O queue. In current implementation we iterate from oldest to most recent pending request and place each one at the head of I/O queue, resulting in reverting requests order (after reschedule the first element on the I/O queue is the oldest deffered request).

Reply:
+ you would eliminate unnecessary list updates, just please deffered list head at the beginning of I/O list and link the old i/O list head at the end of deffered list.